### ISSUE TITLE
question

### ISSUE THREAD

**Issue by twythebest** [wfwf10/Feature-based-Federated-Transfer-Learning#1]
sorrt for bothering you,in you paper:only 1.28x10^(-3)clients join the train every iter,but in your code:you put whole train_loader for iter trainning?I wonder where I can find in your code that means :1.28x10^(-3)clients join the train ?

**Comment by wfwf10** [comment]
Thanks for asking! In the experiment on CIFAR10 - VGG16, we have 50000 training images. We assume there are 6250 clients in total, and each client has 8 different images. In the paper, we said "**each iteration takes a fraction C = 1.28 × 10^(−3) of all clients**". So, in each iteration, we randomly pick "6250 × (1.28 × 10^(−3)) = 8 clients" from all 6250 clients. So, we are randomly picking "8 clients × 8 images/client = 64 images" in each global iteration, and use the average gradient to update the central parameters. In both python files for CIFAR10 - VGG16 experiments, you will find "**batch_size = 64**" in the code.

**Comment by zhaochunyang666** [comment]
sorry to bother you,Why didn't I find the client and server in your code?

**Comment by wfwf10** [comment]
Glad to clarify! In short, **we only save the server parameters, and each batch (with batch_size = 64) includes of all data from clients in one iteration, so terms like "client" and "server" are omitted in the code**. However, you are welcome to add them if needed. 

For all works in this repository, we assume **centralized federated learning (CFL)**. This is different from other architectures such as decentralized federated learning (DFL), where each client may have different parameters. In CFL, during each global iteration, every client uses the same parameter broadcasted from the same server. Therefore, **generating individual gradients from each client and averaging them** is numerically equivalent to **generating gradient from the combined data of all clients**. 

In practice, combining all data into one batch significantly increases computation efficiency (in simulation).
